ExampleMod/Items/Weapons/ExampleGlowSword.cs

```diff
@@ -0,0 +1,49 @@
+using Microsoft.Xna.Framework;
+using Microsoft.Xna.Framework.Graphics;
+using Terraria;
+using Terraria.ID;
+using Terraria.ModLoader;
+
+namespace ExampleMod.Items.Weapons
+{
+	public class ExampleGlowSword : ModItem
+	{
+        //PLEASE DO NOT JUST COPY/PASTE THIS CLASS AND EXPECT YOUR SWORD TO GLOW WHEN SWUNG. In order for that to occur, you need to insert a custom PlayerLayer. See the ModifyDrawLayers method in ExamplePlayer.
+        public override void SetStaticDefaults() 
+		{
+			Tooltip.SetDefault("This sword glows!");
+		}
+
+		public override void SetDefaults() 
+		{
+			item.damage = 50;
+			item.melee = true;
+			item.width = 40;
+			item.height = 40;
+			item.useTime = 20;
+			item.useAnimation = 20;
+			item.useStyle = 1;
+			item.knockBack = 6;
+			item.value = 10000;
+			item.rare = 2;
+			item.UseSound = SoundID.Item1;
+			item.autoReuse = true;
+		}
+
+		public override void AddRecipes() 
+		{
+			ModRecipe recipe = new ModRecipe(mod);
+			recipe.AddIngredient(ItemID.DirtBlock, 10);
+			recipe.AddTile(TileID.WorkBenches);
+			recipe.SetResult(this);
+			recipe.AddRecipe();
+		}
+
+        //This hook lets you draw over your item in the inventory only.
+        public override void PostDrawInInventory(SpriteBatch spriteBatch, Vector2 position, Rectangle frame, Color drawColor, Color itemColor, Vector2 origin, float scale)
```

Comment:
Why aren't you doing this for the animated weapon? And what about drawing in world? (item dropped)